utils/src/main/kotlin/CopyrightStatementsProcessor.kt

```diff
@@ -183,16 +183,20 @@ class CopyrightStatementsProcessor {
         )
     }
 
+    /**
+     * Strip the longest [known copyright prefix][KNOWN_PREFIX_REGEX] from [copyrightStatement] and return a pair of the
+     * copyright statement without the prefix and the prefix that was stripped from it.
+     */
     private fun stripKnownCopyrightPrefix(copyrightStatement: String): Pair<String, String> {
-        val match = KNOWN_PREFIX_REGEX.mapNotNull { regex ->
-            regex.find(copyrightStatement)
-        }.maxByOrNull {
-            it.groups[1]!!.value.length
+        val copyrightStatementWithoutPrefix = KNOWN_PREFIX_REGEX.map { regex ->
+            copyrightStatement.replace(regex, "")
+        }.minByOrNull {
+            it.length
         } ?: return Pair(first = copyrightStatement, second = "")
 
         return Pair(
-            first = copyrightStatement.removeRange(match.groups[1]!!.range),
-            second = match.groups[1]!!.value
+            first = copyrightStatementWithoutPrefix,
+            second = copyrightStatement.removeSuffix(copyrightStatementWithoutPrefix)
```

Comment:
Could you explain why the it is better to avoid the not-null assertions in this particular case. 
It's not obvious to me.

Reply:
In general using the `!!` is error prone as you need to have a deep understanding of the semantics of the code to be able to judge whether it's valid or not. That's why we usually avoid it in ORT's code base wherever possible. Here, its use is also repetitive. And finally, at least for me code with more special characters simply is less pleasant to read than code with fewer special characters. And it looks like I'm [not](https://medium.com/@igorwojda/kotlin-combating-non-null-assertions-5282d7b97205) [alone](https://www.raywenderlich.com/436090-null-safety-tutorial-in-kotlin-best-practices#toc-anchor-012).

Reply:
I've read this linked article as well, it also says "Just to be clear I am not saying not use non-null assertions at all, but rather make 100% sure that you are using this operator correctly and even if you think you are sure then ask on Kotlin slack channel if you really are."

Anyhow, in this particular case I found the previous code to be a bit more strict as it communicates that: It"s for sure this it not null and it's thus sure that the suffix is stripped. The new code says, strip this given suffix in case this suffix exists - otherwise just don't strip anything.

I'd have the preference to not do this change, I'll approve it anyway and leave it up to you.